### PR TITLE
JSM: More TS files for NodeMaterial.

### PR DESCRIPTION
**Pull request by Mugen87**
Follow-up of #17039

### CODE REVIEW

File: examples/jsm/nodes/utils/ColorSpaceNode.d.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+export class ColorSpaceNode extends TempNode {
+  constructor(input: Node, method?: string);
+
+  input: Node;
+  method: string | undefined;
+  nodeType: string;
+
+  generate(builder: NodeBuilder, output: string): string;
+  fromEncoding(encoding: number);
+  fromDecoding(encoding: number);
```
Comment (Mugen87) [inline examples/jsm/nodes/utils/ColorSpaceNode.d.ts]: I guess the parameter of `fromDecoding()` should be actually `decoding`?

File: examples/jsm/nodes/accessors/ColorsNode.d.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { NodeBuilder } from '../core/NodeBuilder';
+import { TempNode } from '../core/TempNode.js';
+
+export class ColorsNode extends TempNode {
```
Comment (Mugen87) [inline examples/jsm/nodes/accessors/ColorsNode.d.ts]: `ColorsNode` has no `nodeType` defined. Is this intended?

File: examples/jsm/nodes/utils/ColorSpaceNode.d.ts (hunk 1 above)
Comment (sunag) [reply examples/jsm/nodes/utils/ColorSpaceNode.d.ts]: I am not sure
this is almost a fork of:
https://github.com/mrdoob/three.js/blob/add00a727b781b3ae1632f52108332edfc28680a/src/renderers/webgl/WebGLProgram.js#L67

And it should use `TextureEncoding`  as input
https://github.com/mrdoob/three.js/blob/5403a8a793e2a64bff1e1f2009f0a120d1db9b48/src/textures/Texture.d.ts#L27

File: examples/jsm/nodes/accessors/ColorsNode.d.ts (hunk 2 above)
Comment (sunag) [reply examples/jsm/nodes/accessors/ColorsNode.d.ts]: > The class ColorsNode has no nodeType defined. Is this intended?

No, no! I must have forgotten to add 👍

File: examples/jsm/nodes/utils/ColorSpaceNode.d.ts (hunk 1 above)
Comment (Mugen87) [reply examples/jsm/nodes/utils/ColorSpaceNode.d.ts]: Okay, then let's keep it for now.

File: examples/jsm/nodes/accessors/ColorsNode.d.ts (hunk 2 above)
Comment (Mugen87) [reply examples/jsm/nodes/accessors/ColorsNode.d.ts]: Okay, we can resolve that later 😉

**Comment by Mugen87** [comment]
Okay, the type declaration files for `NodeMaterial` are complete. We now have the chance to polish them^^.

**Comment by sunag** [comment]
This is great 👏 👏 👏 ! Thanks.

**Comment by mrdoob** [comment]
Many thanks!

**Pull request merged**